src/main/java/io/lettuce/core/ReadFrom.java

```diff
@@ -110,6 +112,29 @@
      */
     public static final ReadFrom ANY_REPLICA = new ReadFromImpl.ReadFromAnyReplica();
 
+    /**
+     * Setting to read from any node in the subnets.
+     *
+     * @param cidrNotations CIDR-block notation strings, e.g., "192.168.0.0/16", "2001:db8:abcd:0000::/52". Must not be
+     *        {@code null}.
+     * @return an instance of {@link ReadFromImpl.ReadFromSubnet}.
+     * @since x.x.x
+     */
+    public static ReadFrom subnet(String... cidrNotations) {
+        return new ReadFromImpl.ReadFromSubnet(cidrNotations);
+    }
+
+    /**
+     * Read from any node that has {@link RedisURI} matching with the given pattern.
+     *
+     * @param pattern regex pattern, e.g., {@code Pattern.compile(".*region-1.*")}. Must not be {@code null}.
+     * @return an instance of {@link ReadFromImpl.ReadFromRegex}.
+     * @since x.x.x
+     */
+    public static ReadFrom regex(Pattern pattern) {
+        return new ReadFromImpl.ReadFromRegex(pattern);
+    }
```

Comment:
For hostname-based matching, I feel regex-based is more in demand than set-based because it is more simple and extensible, but what do you think? If not I will replace to set-based as originally proposed.

Reply:
I think that's fine, too.